langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Find.java

```diff
@@ -74,9 +94,14 @@ public static BArray findAll(BRegexpValue regExp, BString str, int startIndex) {
         return resultArray;
     }
 
-    public static BArray findAllGroups(BRegexpValue regExp, BString str, int startIndex) {
-        Matcher matcher = RegexUtil.getMatcher(regExp, str);
-        matcher.region(startIndex, str.length());
+    public static BArray findAllGroups(BRegexpValue regExp, BString str, long startIndex) {
+        checkIndexWithinRange(str, startIndex);
+        if (isEmptyStrOrRegexp(regExp, str)) {
+            return null;
```

Comment:
Lets cover this

Reply:
Added in https://github.com/ballerina-platform/ballerina-lang/pull/39444/commits/1f7c4326bad3eda9e064c2de7b125b53e11ce284.